### PR TITLE
Add installation registration producer AGPUSH-2101

### PR DESCRIPTION
**Pull request by dimitraz**
Implementation of the first Kafka Producer in the `InstallationRegistrationEndpoint`.
More details: [AGPUSH-2101 ](https://issues.jboss.org/browse/AGPUSH-2101 )

### CODE REVIEW

**Comment by dimitraz** [comment]
Please review @matzew @ziccardi @polinankoleva

File: jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java [hunk 1]
```diff
@@ -217,17 +223,28 @@ public Response registerInstallation(
...
 
         // find the matching variation:
         final Variant variant = loadVariantWhenAuthorized(request);
         if (variant == null) {
             return create401Response(request);
         }
 
-        //let's do update the analytics
         if (pushMessageId != null) {
+            try (InputStream props = this.getClass().getResourceAsStream("/kafka/producer.props")) {
```
Comment (matzew) [inline jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java]: I think for now embedding resource bundles, w/ hard-coded values are OK.

I'd like to move body of code out to a helper class, e.g. in the `common` module.

File: jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java [hunk 2]
```diff
@@ -51,8 +54,11 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.ResponseBuilder;
 import javax.ws.rs.core.Response.Status;
+import java.io.FileInputStream;
```
Comment (matzew) [inline jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java]: the build fails w/ a check-style error:

```
[ERROR] InstallationRegistrationEndpoint.java[57:8] Unused import - java.io.FileInputStream.
```

File: jaxrs/src/main/resources/kafka/producer.props [hunk 3]
```diff
@@ -0,0 +1,9 @@
+bootstrap.servers=172.19.0.3:9092
+acks=all
+retries=0
+batch.size=16384
+auto.commit.interval.ms=1000
+linger.ms=1
+buffer.memory=33554432
+key.serializer=org.apache.kafka.common.serialization.StringSerializer
+value.serializer=org.apache.kafka.common.
```
Comment (matzew) [inline jaxrs/src/main/resources/kafka/producer.props]: is there something missing here, on the last line ?

File: jaxrs/src/main/resources/kafka/producer.props [hunk 4]
```diff
@@ -0,0 +1,9 @@
+bootstrap.servers=172.19.0.3:9092
```
Comment (matzew) [inline jaxrs/src/main/resources/kafka/producer.props]: As said, for now this is fine.

However, version 0.0.5 of the kafka-cdi-extension is able to resolve simple expressions, like:
```
#{HOST}:#{PORT}"
```

Perhaps its a good idea to look at `javax.el` APIs for this. See AGPUSH-2104

File: jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java [hunk 5]
```diff
@@ -217,17 +222,28 @@ public Response registerInstallation(
...
         // find the matching variation:
         final Variant variant = loadVariantWhenAuthorized(request);
         if (variant == null) {
             return create401Response(request);
         }
 
-        //let's do update the analytics
         if (pushMessageId != null) {
+            try (InputStream props = this.getClass().getResourceAsStream("/kafka/producer.props")) {
+                Properties properties = new Properties();
```
Comment (matzew) [inline jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java]: I think we should use the `final` keyword here, to indicate that the `Properties` and `Producer` are immutable

Comment (matzew) [inline jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java]: Will add that, when merging ;-)

File: jaxrs/src/main/resources/kafka/producer.props (hunk 4 above)
Comment (dimitraz) [inline jaxrs/src/main/resources/kafka/producer.props]: Agreed, just a quick fix for now but we will definitely have to address it in `AGPUSH-2104`

File: jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java (hunk 5 above)
Comment (dimitraz) [inline jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java]: Ok thanks :)

**Comment by matzew** [comment]
merged and edited as discussed!

**Pull request merged**